jenkins-master/jobs/scripts/workspace/runtests.py

```diff
@@ -60,12 +65,16 @@ def run(self):
         # Set environment variable to let mozcrash save a copy of the minidump files
         env.update({'MINIDUMP_SAVE_PATH': os.path.join(here, 'minidumps')})
 
-        command = [sys.executable,
+        command = [sys.executable, '-u',
                    os.path.join('mozharness', 'scripts', self.settings['harness_script'])]
         command.extend(self.query_args())
 
-        print('Calling command to execute tests: {}'.format(command))
-        return subprocess.call(command, env=env)
+        logger.info('Calling command to execute tests: {}'.format(command))
+        try:
+            return subprocess.check_call(command, env=env)
+        except BaseException as e:
+            logger.exception('Failed to run external process')
```

Comment:
> Shouldn't there be some logging of the text of the exception here?

This is not necessary. `logger.exception()` is automatically adding the traceback information on its own.

> Why BaseException instead of Exception or CalledProcessError? Do you really want to silence KeyboardInterrupt?

We don't silence those. The full traceback would be visible and we definitely would have a retval.ext file written. But I see the problem now... even right now we don't use `usercancel` as result but always `busted`. So we should definitely handle that separately.